p4rt_app/utils/table_utility.cc

```diff
@@ -15,43 +15,89 @@
 
 #include "absl/status/status.h"
 #include "gutil/status.h"
+#include "absl/strings/substitute.h"
 #include "p4_pdpi/utils/annotation_parser.h"
 #include "swss/schema.h"
 
 namespace p4rt_app {
 namespace table {
 
+// This is what orch agent understands as fixed tables
+std::vector<std::string> FixedTables {
+  APP_P4RT_ROUTER_INTERFACE_TABLE_NAME,
+  APP_P4RT_NEIGHBOR_TABLE_NAME,
+  APP_P4RT_NEXTHOP_TABLE_NAME,
+  APP_P4RT_WCMP_GROUP_TABLE_NAME,
+  APP_P4RT_IPV4_TABLE_NAME,
+  APP_P4RT_IPV6_TABLE_NAME,
+  APP_P4RT_MIRROR_SESSION_TABLE_NAME,
+  APP_P4RT_L3_ADMIT_TABLE_NAME,
+  APP_P4RT_TUNNEL_TABLE_NAME
+};
+
 std::string TypeName(Type type) {
   switch (type) {
     case Type::kAcl:
       return "ACL";
     case Type::kFixed:
       return "FIXED";
-    case Type::kDefinition:
+    case Type::kExt:
+      return "EXT";
+    case Type::kAclDefinition:
       return APP_P4RT_ACL_TABLE_DEFINITION_NAME;
+    case Type::kTblsDefinitionSet:
+      return APP_P4RT_TABLES_DEFINITION_TABLE_NAME;
   }
   // TODO: return status failure.
 }
 
 absl::StatusOr<Type> TypeParse(absl::string_view type_name) {
   if (type_name == "ACL") return Type::kAcl;
   if (type_name == "FIXED") return Type::kFixed;
-  if (type_name == APP_P4RT_ACL_TABLE_DEFINITION_NAME) return Type::kDefinition;
+  if (type_name == "EXT") return Type::kExt;
+  if (type_name == APP_P4RT_ACL_TABLE_DEFINITION_NAME) {
+          return Type::kAclDefinition;
+  }
+  if (type_name == APP_P4RT_TABLES_DEFINITION_TABLE_NAME) {
+          return Type::kTblsDefinitionSet;
+  }
   return gutil::InvalidArgumentErrorBuilder()
          << "\"" << type_name << "\" does not name a valid table::Type.";
 }
 
 }  // namespace table
 
+table::Type conditional_kExt(const pdpi::IrTableDefinition& ir_table)
+{
+    if (ir_table.preamble().id() > 0x03000000)
+    {
+        return table::Type::kExt;
+    }
+    return table::Type::kFixed;
```

Comment:
Given we have already checked valid fixed tables just before this, this is not required here.

I would suggest to move this code inline at the caller place and return InvalidArgument if table_id is not greater than 0x030000.